flight_statistics.py

```
@@ -0,0 +1,528 @@
+                "GROUP BY label ORDER BY total DESC, label ASC LIMIT ?",
+                (*params, limit),
+            ).fetchall()
+        ]
+
+    @staticmethod
+    def _repeat(connection, where, params):
+        row = connection.execute(
+            "SELECT COALESCE(MAX(registration), MAX(callsign), MAX(hex), identity) "
+            "AS label, COUNT(*) AS total FROM flight_encounters "
+            f"WHERE {where} GROUP BY identity "
+            "ORDER BY total DESC, label ASC LIMIT 1",
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)